lib/modules/tmux.nix

```diff
@@ -0,0 +1,26 @@
+{ lib }:
+let
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    types
+    ;
+
+in
+{
+  # TODO: Bring this from home-manager!
+  pluginWithConfigModule = types.submodule {
```

Comment:
It feels like [Deja Vu](https://www.youtube.com/watch?v=dv13gl0a-FA). Should we maybe factor that into a generic `pluginWithConfigModule`?

Reply:
Yes I thought of that. but I think it's better (IMO) to just inherit upstream definitions somehow. I left it as a TODO for now to focus on functionality.